core-container/entry.sh

```diff
@@ -23,145 +23,123 @@ function echoerr() {
    echo "ERROR: $1" >&2
 }
 
-
-function set_provider_adapters() {
-    oldKubeconfig=$KUBECONFIG
-    KUBECONFIG=
-    echo "Waiting for space manager to be ready... this may take a while."
+function wait_kcp_ready() {
+    echo "Waiting for kcp to be ready... this may take a while."
     (
-        until [ "$(kubectl logs $(kubectl get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c space-manager | grep '***READY***')" != "" ]; do
-            sleep 10
+        until [ "$(kubectl --kubeconfig $kubestellar_kubeconfig logs $(kubectl --kubeconfig $kubestellar_kubeconfig get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c kcp | grep '***READY***')" != "" ]; do
+           sleep 10
         done
     )
+}
 
-    echo "Delete the kcp provider secret if it already exists, and then create it."
-    if ! kubectl delete secret -n ${NAMESPACE} kcpsec ; then
-        echo "Nothing to delete."
-    fi
-    kubectl get secrets kubestellar -o 'go-template={{index .data "admin.kubeconfig"}}' | base64 --decode > kcpsecret
-    kubectl create secret generic -n ${NAMESPACE} kcpsec --from-file=kubeconfig="kcpsecret"    
-    rm kcpsecret
+function get_kcp_kubeconfig() {
+    wait_kcp_ready
```

Comment:
That's a great question.  This is actually old logic flow that I kept, but revisiting it I see we can remove the wait_kcp_ready since the subsequent check for the existance of the kubestellar secret is actually sufficient.